batch/batch/cloud/gcp/driver/create_instance.py

```diff
@@ -234,62 +208,60 @@ def scheduling() -> dict:
 
 sudo mkdir -p /etc/netns
 
-# Setup fluentd
+# Setup ops agent
 touch /worker.log
 touch /run.log
+mkdir -p /batch/jvm-container-logs/
 
-sudo rm /etc/google-fluentd/config.d/*  # remove unused config files
-
-sudo tee /etc/google-fluentd/config.d/worker-log.conf <<EOF
-<source>
-@type tail
-format json
-path /worker.log
-pos_file /var/lib/google-fluentd/pos/worker-log.pos
-read_from_head true
-tag worker.log
-</source>
-
-<filter worker.log>
-@type record_transformer
-enable_ruby
-<record>
-    severity \${{ record["levelname"] }}
-    timestamp \${{ record["asctime"] }}
-</record>
-</filter>
-EOF
-
-sudo tee /etc/google-fluentd/config.d/run-log.conf <<EOF
-<source>
-@type tail
-format none
-path /run.log
-pos_file /var/lib/google-fluentd/pos/run-log.pos
-read_from_head true
-tag run.log
-</source>
-EOF
-
-sudo tee /etc/google-fluentd/config.d/jvm-logs.conf <<EOF
-{jvm_fluentd_config}
-EOF
-
-sudo cp /etc/google-fluentd/google-fluentd.conf /etc/google-fluentd/google-fluentd.conf.bak
-head -n -1 /etc/google-fluentd/google-fluentd.conf.bak | sudo tee /etc/google-fluentd/google-fluentd.conf
-sudo tee -a /etc/google-fluentd/google-fluentd.conf <<EOF
-labels {{
-"namespace": "$NAMESPACE",
-"instance_id": "$INSTANCE_ID"
-}}
-</match>
+sudo tee /etc/google-cloud-ops-agent/config.yaml <<EOF
+logging:
+  receivers:
+    runlog:
+      type: files
+      include_paths:
+      - /run.log
+      record_log_file_path: true
+    workerlog:
+      type: files
+      include_paths:
+      - /worker.log
+      record_log_file_path: true
+    jvmlog:
+      type: files
+      include_paths:
+      - /batch/jvm-container-logs/jvm-*.log
+      record_log_file_path: true
+  processors:
+    labels:
+      type: modify_fields
+      fields:
+        labels.namespace:
+          static_value: $NAMESPACE
+        labels.instance_id:
+          static_value: $INSTANCE_ID
+  service:
+    pipelines:
+      default_pipeline:
+        processors: [labels]
+        receivers: [runlog, workerlog, jvmlog]
+
+metrics:
+  receivers:
+    hostmetrics:
+      type: hostmetrics
+      collection_interval: 60s
+  processors:
+    metrics_filter:
+      type: exclude_metrics
+      metrics_pattern: []
+  service:
+    pipelines:
+      default_pipeline:
+        receivers: [hostmetrics]
+        processors: [metrics_filter]
 EOF
-rm /etc/google-fluentd/google-fluentd.conf.bak
 
-mkdir -p /batch/jvm-container-logs/
-{jvm_touch_command}
-
-sudo service google-fluentd restart
+sudo systemctl restart google-cloud-ops-agent
```

Comment:
Since we restart google-fluentd currently, I'll approve this, but I'm gonna create an issue to (1) characterize worker startup time and (2) reduce it.

Reply:
The reason for doing the restart is we can't know the namespace and instance name ahead of time and have to modify the config anyways to insert these labels.

Reply:
The issue in question: https://github.com/hail-is/hail/issues/13976